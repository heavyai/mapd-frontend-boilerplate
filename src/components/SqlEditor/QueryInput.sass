.container
  position: relative
  padding: 10px 18px 18px

.input
  width: 100%

.progress
  position: absolute
  height: 3px
  bottom: 25px
