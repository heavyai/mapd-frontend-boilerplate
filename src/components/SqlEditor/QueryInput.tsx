import React from 'react'
import { TextField } from 'rmwc/TextField'
import LinearProgress from 'rmwc/LinearProgress'

import { Dispatcher } from '~/store'
import { setStatement, sendQuery } from '~/actions/query'

const styles = require( './QueryInput.sass' )


type Props = {
  dispatch: Dispatcher,
  statement: string,
  pending: boolean
}

const QueryInput: React.StatelessComponent<Props> = ( { dispatch, statement, pending } ) => {
  const handleInputChange = ( event: React.FormEvent<HTMLInputElement> ) => {
    dispatch( setStatement( event.currentTarget.value ) )
  }

  const handleInputKeyPress = ( event: React.KeyboardEvent<HTMLInputElement> ) => {
    if ( event.key === 'Enter' ) {
      dispatch( sendQuery.action( statement ) )
    }
  }

  return (
    <div className={ styles.container }>
      <TextField
        className={ styles.input }
        label='Enter a SQL statement'
        spellCheck={ false }
        value={ statement }
        onChange={ handleInputChange }
        onKeyPress={ handleInputKeyPress }
      />
      { pending && <LinearProgress className={ styles.progress } determinate={ false } /> }
    </div>
  )
}


export default QueryInput
