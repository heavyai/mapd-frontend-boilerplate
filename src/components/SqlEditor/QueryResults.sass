.container
  padding: 0 18px 16px

.table
  border-spacing: 0

  th
    text-align: initial
    padding-bottom: 12px

  td
    padding: 0 3px 3px 0

.error
  color: red

.noResults
  color: #ccc
