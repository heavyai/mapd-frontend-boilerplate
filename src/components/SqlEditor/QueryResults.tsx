import React from 'react'

const styles = require( './QueryResults.sass' )


import { QueryRequest } from '~/reducers/query'

type Props = {
  request: QueryRequest
}


const table = ( columns: string[], data: Object[] ) =>
  <table className={ styles.table }>
    <thead>
      <tr>
        { columns.map( ( column ) => <th key={ column }>{ column }</th> ) }
      </tr>
    </thead>
    <tbody>
      {
        data.map( ( row, index ) =>
          <tr key={ index }>
            {
              columns.map( ( column ) =>
                <td key={ `${ column }-${ index }` }>{ row[ column ] }</td>
              )
            }
          </tr>
        )
      }
    </tbody>
  </table>


const QueryResults: React.StatelessComponent<Props> = ( { request: { results, error } } ) =>
  <div className={ styles.container }>
    { error
      ? <div className={ styles.error }>{ error.message }</div>
      : results && results.length
        ? table( Object.keys( results[ 0 ] ), results )
        : <div className={ styles.noResults }>No results</div>
    }
  </div>


export default QueryResults
