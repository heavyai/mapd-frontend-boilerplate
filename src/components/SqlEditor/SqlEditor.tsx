import React from 'react'

import { Dispatcher } from '~/store'
import { QueryState } from '~/reducers/query'
import QueryInput from '~/components/SqlEditor/QueryInput'
import QueryResults from '~/components/SqlEditor/QueryResults'

const styles = require( './SqlEditor.sass' )


type Props = {
  dispatch: Dispatcher,
  query: QueryState
}

const SqlEditor: React.StatelessComponent<Props> = ( { dispatch, query: { statement, request } } ) =>
  <div className={ styles.container }>
    <QueryInput dispatch={ dispatch } statement={ statement } pending={ request.pending } />
    { request.submitted && <QueryResults request={ request } /> }
  </div>


export default SqlEditor
