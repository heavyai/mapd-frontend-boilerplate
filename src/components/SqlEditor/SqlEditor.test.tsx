import React from 'react'
import { render } from 'enzyme'

import SqlEditor from './SqlEditor'


test( 'Component renders', () => {
  const component = render(
    <SqlEditor
      dispatch={ jest.fn() }
      query={ { statement: '', request: { submitted: false, pending: false } } }
    />
  )

  expect( component ).toMatchSnapshot()
} )
