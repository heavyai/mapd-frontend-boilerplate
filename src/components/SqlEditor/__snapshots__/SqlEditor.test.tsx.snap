// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
<div
  class="container"
>
  <div
    class="container"
  >
    <div
      class="input   mdc-text-field mdc-text-field--upgraded"
    >
      <input
        class=" mdc-text-field__input"
        id="text-field-test"
        spellcheck="false"
        type="text"
        value=""
      />
      <label
        class=" mdc-floating-label"
        for="text-field-test"
      >
        Enter a SQL statement
      </label>
      <div
        class=" mdc-line-ripple"
      />
    </div>
  </div>
</div>
`;
